src/index.ts

```diff
@@ -173,6 +173,9 @@ export default class HDKeyring implements Keyring<SerializedHDKeyring> {
   }
 
   addAddressesSync(numNewAccounts = 1): string[] {
+    if (numNewAccounts < 0 || numNewAccounts > 2 ** 31 - 1) {
```

Comment:
Is it numNewAccounts or numNewAccounts + numAddresses that must not exceed 2**31 - 1?

Reply:
The real limit is the total accounts — I realized after this landed and spaced, good catch.